crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs

```diff
@@ -430,7 +430,30 @@ impl Analyzer<'_, '_> {
                     }
                     _ => {}
                 }
-                callee_ty
+
+                match callee_ty.normalize() {
+                    Type::Union(u) => {
+                        let types = u
+                            .types
+                            .iter()
+                            .cloned()
+                            .filter(|callee| !matches!(callee.normalize(), Type::Module(..)))
```

Comment:
Also, please check `Type::Namespace` too.

`module Foo` is `Type::Namespace`, and `namespace Foo` is `Type::Module`